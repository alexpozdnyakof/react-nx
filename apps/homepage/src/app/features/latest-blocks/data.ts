import { BlockShort } from '@blockchain/api-interfaces';

export const BLX_DATA: Array<BlockShort> = [
  ...new Array(5).fill({
    index: 75260722,
    time: Date.now(),
    size: 1200,
    txsCount: 723,
    occupancy: 71,
  }),
];
