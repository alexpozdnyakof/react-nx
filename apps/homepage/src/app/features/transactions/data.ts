export const DEMO_DATA = [
  {
    hash: '123-qwe',
    time: 1663353730855,
    amountUsd: 1000,
    amountBtc: 1000,
  },
  {
    hash: '123-qwe',
    time: 1663353730855,
    amountUsd: 1000,
    amountBtc: 1000,
  },
  {
    hash: '123-qwe',
    time: 1663353730855,
    amountUsd: 1000,
    amountBtc: 1000,
  },
];
