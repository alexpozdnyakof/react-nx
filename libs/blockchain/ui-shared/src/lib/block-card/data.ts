const BLX_DATA = [
  {
    index: '752,722',
    time: new Intl.DateTimeFormat('en-US', {
      year: 'numeric',
      month: 'numeric',
      day: 'numeric',
      hour: 'numeric',
      minute: 'numeric',
      second: 'numeric',
      hour12: false,
    }).format(Date.now()),
    size: '1.2 Mb',
    txsCount: 723,
    occupancy: 71,
  },
];

export { BLX_DATA };
