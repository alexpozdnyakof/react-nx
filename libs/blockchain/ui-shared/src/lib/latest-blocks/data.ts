const BLXS_DATA = [
  ...new Array(5).fill({
    index: '752,722',
    time: new Intl.DateTimeFormat('en-US', {
      year: 'numeric',
      month: 'numeric',
      day: 'numeric',
      hour: 'numeric',
      minute: 'numeric',
      second: 'numeric',
      hour12: false,
    }).format(Date.now()),
    size: '1.2 Mb',
    txsCount: 723,
    occupancy: '71%',
  }),
];
export { BLXS_DATA };
