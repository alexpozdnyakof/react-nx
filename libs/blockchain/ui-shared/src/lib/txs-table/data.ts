const TXS_DATA = [
  ...new Array(10).fill(['ccf3e-2cd36', Date.now(), 0.76729409, 15321.09]),
].map(([hash, time, amountBtc, amountUsd]) => ({
  hash,
  time,
  amountBtc,
  amountUsd,
}));

export { TXS_DATA };
