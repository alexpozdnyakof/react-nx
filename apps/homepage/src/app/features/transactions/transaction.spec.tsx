import { configureStore } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { DEMO_DATA } from './data';
import reducer, { load, loadSuccess } from './slice';
import LatestTransactions from './transactions';
import testRenderer from 'react-test-renderer'
import * as storeHooks from '../../store-hooks'

describe('latest transactions', () => {
  const store = configureStore({reducer: {latestTxs: reducer}});
  const WithStore = () => (<Provider store={store}><LatestTransactions/></Provider>);

  it('should render succesfully', () => {
    const base = render(<WithStore/>)
    expect(base).toBeDefined()
  })

  it('should render correctly with transactions', ()=>{
    store.dispatch(loadSuccess(DEMO_DATA))
    const tree = testRenderer.create(<WithStore/>).toJSON();

    expect(tree).toMatchSnapshot();
  })

  it('should dispatch load', ()=>{
    const dispatchMock = jest.fn()
    jest.spyOn(storeHooks, 'useAppDispatch').mockReturnValue(dispatchMock);

    render(<WithStore/>);
    expect(dispatchMock).toHaveBeenCalledWith(load());
  })
})
