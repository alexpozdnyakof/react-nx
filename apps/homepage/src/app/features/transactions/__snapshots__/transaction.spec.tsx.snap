// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`latest transactions should render correctly 1`] = `
<div
  className="Widget"
>
  <div
    className="Widget-Header"
  >
    <img
      alt="widget symbol"
      className="Widget-Symbol"
      src="https://www.blockchain.com/explorer/_next/static/media/btc.a6006067.png"
    />
    <div
      className="Widget-Info"
    >
      <a
        className="Link"
        data-testid="title-link"
        href="https://www.blockchain.com/btc/unconfirmed-transactions"
      >
        <h1
          className="Widget-Title"
        >
          Latest Transactions
        </h1>
      </a>
      <h2
        className="Widget-SubTitle"
      >
        Bitcoin
      </h2>
    </div>
    <div
      className="Spacer"
    />
    <a
      className="Link"
      data-testid="arrow-link"
      href="https://www.blockchain.com/btc/unconfirmed-transactions"
    >
      <span
        className="Widget-Arrow"
      >
        →
      </span>
    </a>
  </div>
  <div
    className="Table"
  >
    <div
      className="Table-Row"
      data-testid="table-body-row"
    >
      <span
        className="Tx-Cell"
      >
        123-qwe
      </span>
      <span
        className="Tx-Cell Tx-Cell_time"
      >
        17/09/2022
      </span>
      <span
        className="Tx-Cell Tx-Cell_amount"
      >
        1000 Btc
         BTC
      </span>
      <span
        className="Tx-Cell Tx-Cell_amount"
      >
        1000 Usd
      </span>
    </div>
    <div
      className="Table-Row"
      data-testid="table-body-row"
    >
      <span
        className="Tx-Cell"
      >
        123-qwe
      </span>
      <span
        className="Tx-Cell Tx-Cell_time"
      >
        17/09/2022
      </span>
      <span
        className="Tx-Cell Tx-Cell_amount"
      >
        1000 Btc
         BTC
      </span>
      <span
        className="Tx-Cell Tx-Cell_amount"
      >
        1000 Usd
      </span>
    </div>
    <div
      className="Table-Row"
      data-testid="table-body-row"
    >
      <span
        className="Tx-Cell"
      >
        123-qwe
      </span>
      <span
        className="Tx-Cell Tx-Cell_time"
      >
        17/09/2022
      </span>
      <span
        className="Tx-Cell Tx-Cell_amount"
      >
        1000 Btc
         BTC
      </span>
      <span
        className="Tx-Cell Tx-Cell_amount"
      >
        1000 Usd
      </span>
    </div>
  </div>
</div>
`;

exports[`latest transactions should render correctly with transactions 1`] = `
<div
  className="Widget"
>
  <div
    className="Widget-Header"
  >
    <img
      alt="widget symbol"
      className="Widget-Symbol"
      src="https://www.blockchain.com/explorer/_next/static/media/btc.a6006067.png"
    />
    <div
      className="Widget-Info"
    >
      <a
        className="Link"
        data-testid="title-link"
        href="https://www.blockchain.com/btc/unconfirmed-transactions"
      >
        <h1
          className="Widget-Title"
        >
          Latest Transactions
        </h1>
      </a>
      <h2
        className="Widget-SubTitle"
      >
        Bitcoin
      </h2>
    </div>
    <div
      className="Spacer"
    />
    <a
      className="Link"
      data-testid="arrow-link"
      href="https://www.blockchain.com/btc/unconfirmed-transactions"
    >
      <span
        className="Widget-Arrow"
      >
        →
      </span>
    </a>
  </div>
  <div
    className="Table"
  >
    <div
      className="Table-Row"
      data-testid="table-body-row"
    >
      <span
        className="Tx-Cell"
      >
        123-qwe
      </span>
      <span
        className="Tx-Cell Tx-Cell_time"
      >
        17/09/2022
      </span>
      <span
        className="Tx-Cell Tx-Cell_amount"
      >
        1000 Btc
         BTC
      </span>
      <span
        className="Tx-Cell Tx-Cell_amount"
      >
        1000 Usd
      </span>
    </div>
    <div
      className="Table-Row"
      data-testid="table-body-row"
    >
      <span
        className="Tx-Cell"
      >
        123-qwe
      </span>
      <span
        className="Tx-Cell Tx-Cell_time"
      >
        17/09/2022
      </span>
      <span
        className="Tx-Cell Tx-Cell_amount"
      >
        1000 Btc
         BTC
      </span>
      <span
        className="Tx-Cell Tx-Cell_amount"
      >
        1000 Usd
      </span>
    </div>
    <div
      className="Table-Row"
      data-testid="table-body-row"
    >
      <span
        className="Tx-Cell"
      >
        123-qwe
      </span>
      <span
        className="Tx-Cell Tx-Cell_time"
      >
        17/09/2022
      </span>
      <span
        className="Tx-Cell Tx-Cell_amount"
      >
        1000 Btc
         BTC
      </span>
      <span
        className="Tx-Cell Tx-Cell_amount"
      >
        1000 Usd
      </span>
    </div>
  </div>
</div>
`;
